git-clone/main.tf

```diff
@@ -26,19 +26,46 @@ variable "agent_id" {
 }
 
 locals {
-  clone_path = var.base_dir != "" ? join("/", [var.base_dir, replace(basename(var.url), ".git", "")]) : join("/", ["~", replace(basename(var.url), ".git", "")])
+  # Remove query parameters and branch name from the URL
+  url = replace(replace(var.url, "/\\?.*/", ""), "/#.*", "")
+  # Remove tree and branch name from the URL
+  clone_url = replace(replace(local.url, "//-/tree/.*/", ""), "//tree/.*/", "")
+  # Extract the branch name from the URL
+  branch_name = replace(replace(replace(local.url, local.clone_url, ""), "/.*/-/tree//", ""), "/.*/tree//", "")
+  # Construct the path to clone the repository
+  clone_path = var.base_dir != "" ? join("/", [var.base_dir, replace(basename(local.clone_url), ".git", "")]) : join("/", ["~", replace(basename(local.clone_url), ".git", "")])
+
+  # git@gitlab.com:mike.brew/repo-tests.log.git
+  # becomes https://gitlab.com/mike.brew/repo-tests.log.git
+  web_url = startswith(local.clone_url, "git@") ? replace(replace(local.clone_url, ":", "/"), "git@", "https://") : local.clone_url
 }
 
 output "repo_dir" {
   value       = local.clone_path
   description = "Full path of cloned repo directory"
 }
 
+output "clone_url" {
+  value       = local.clone_url
+  description = "Git repository URL"
+}
+
+output "web_url" {
+  value       = local.web_url
+  description = "Git https repository URL"
```

Comment:
```suggestion
  description = "Git HTTPS repository URL (may be invalid for unsupported providers)"
```

Reply:
FYI, for context the `web_url` would be used to add a `coder_app` to open the related repo:

<img width="1256" alt="Screenshot 2024-04-05 at 9 18 04 AM" src="https://github.com/coder/modules/assets/5442469/89babc22-9980-4a06-afc3-5b1ae14edd3d">

I can share a complete example